.generator/src/generator/templates/api.j2

```diff
@@ -300,18 +307,41 @@ impl {{ structName }} {
         {%- endif %}
         {%- endfor %}
 
+        // build headers
+        let mut headers = HeaderMap::new();
+        {%- if operation.requestBody %}
+        {%- set contentTypeHeaders = operation.requestBody.content.keys()|list %}
+        {%- if contentTypeHeaders %}
+        {%- if "application/json" in contentTypeHeaders %}
+        headers.insert("Content-Type", "application/json".parse().unwrap());
+        {%- else %}
+        headers.insert("Content-Type", "{{ contentTypeHeaders[0] }}".parse().expect("failed to parse Content-Type header"));
+        {%- endif %}
+        {%- endif %}
+        {%- endif %}
+        {%- if operation.responses %}
+        {%- set acceptHeaders = operation|accept_headers %}
+        {%- if acceptHeaders %}
+        {%- if "application/json" in acceptHeaders %}
+        headers.insert("Accept", "application/json".parse().unwrap());
+        {%- else  %}
+        headers.insert("Accept", "{{ acceptHeaders|join(",") }}".parse().expect("failed to parse Accept header"));
+        {%- endif %}
+        {%- endif %}
+        {%- endif %}
+
         {% for name, parameter in operation|parameters if parameter.in == "header" %}
         {%- if not parameter.required %}
         if let Some(ref local) = {{name|variable_name}} {
-            local_req_builder = local_req_builder.header("{{name}}", &local.to_string());
-        };
+            headers.insert("{{name}}", local.to_string().parse().expect("failed to parse {{name}} header"));
+        }
         {%- else %}
-        local_req_builder = local_req_builder.header("{{name}}", &{{name|variable_name}}.to_string());
+        headers.insert("{{name}}", {{name|variable_name}}.to_string().parse().expect("failed to parse {{name}} header"));
         {%- endif %}
         {%- endfor %}
 
         // build user agent
-        local_req_builder = local_req_builder.header(reqwest::header::USER_AGENT, local_configuration.user_agent.clone());
+        headers.insert(reqwest::header::USER_AGENT, local_configuration.user_agent.clone().parse().expect("failed to parse User Agent header"));
```

Comment:
We should make the user agent configurable and probably not fail if its missing or invalid.

Reply:
changed to fallback to the default user agent